### PR TITLE
[Storehouse] Optimize finalized reader to use block id index

### PR DESCRIPTION
**Pull request by zhangchiqing**
This PR optimizes the FinalizedBlockIDAtHeight method of the FinalizedReader used by Storehouse

### CODE REVIEW

File: engine/execution/ingestion/loader/unexecuted_loader.go [hunk 1]
```diff
@@ -231,8 +231,8 @@ func (e *UnexecutedLoader) pendingUnexecutedBlocks(ctx context.Context, finalize
 // if the EN is dynamically bootstrapped, the finalized blocks at height range:
 // [ sealedRoot.Height, finalizedRoot.Height - 1] can not be retrieved from
 // protocol state, but only from headers
-func (e *UnexecutedLoader) getHeaderByHeight(height uint64) (*flow.Header, error) {
+func (e *UnexecutedLoader) getHeaderByHeight(height uint64) (flow.Identifier, error) {
```
Comment (jordanschalm) [inline engine/execution/ingestion/loader/unexecuted_loader.go:234]:
```suggestion
func (e *UnexecutedLoader) getBlockIDByHeight(height uint64) (flow.Identifier, error) {
```

File: engine/execution/ingestion/loader/unfinalized_loader.go [hunk 2]
```diff
@@ -89,8 +89,8 @@ func (e *UnfinalizedLoader) LoadUnexecuted(ctx context.Context) ([]flow.Identifi
 // if the EN is dynamically bootstrapped, the finalized blocks at height range:
 // [ sealedRoot.Height, finalizedRoot.Height - 1] can not be retrieved from
 // protocol state, but only from headers
-func (e *UnfinalizedLoader) getHeaderByHeight(height uint64) (*flow.Header, error) {
+func (e *UnfinalizedLoader) getHeaderByHeight(height uint64) (flow.Identifier, error) {
```
Comment (jordanschalm) [inline engine/execution/ingestion/loader/unfinalized_loader.go:92]:
```suggestion
func (e *UnfinalizedLoader) getBlockIDByHeight(height uint64) (flow.Identifier, error) {
```

**Review by jordanschalm — APPROVED** [review]

File: engine/execution/ingestion/loader/unfinalized_loader.go [hunk 3]
```diff
@@ -60,12 +60,12 @@ func (e *UnfinalizedLoader) LoadUnexecuted(ctx context.Context) ([]flow.Identifi
 	// reload its block to execution queues
 	// loading finalized blocks
 	for height := lastExecuted + 1; height <= final.Height; height++ {
-		header, err := e.getHeaderByHeight(height)
+		finalizedID, err := e.getHeaderByHeight(height)
```
Comment (peterargue) [inline engine/execution/ingestion/loader/unfinalized_loader.go:63]: why not use `e.headers.BlockIDByHeight` directly?

File: engine/execution/ingestion/loader/unexecuted_loader.go (hunk 1 above)
Comment (peterargue) [reply engine/execution/ingestion/loader/unexecuted_loader.go:234]: same question here about not using `e.headers.BlockIDByHeight(height)` directly

**Review by peterargue — APPROVED** [review]

**Pull request merged**